StyleCop.Analyzers/StyleCop.Analyzers.Test/OrderingRules/SA1204UnitTests.cs

```diff
@@ -22,16 +22,66 @@ public class SA1204UnitTests : DiagnosticVerifier
         public async Task TestValidOrderingAsync()
         {
             var testCode = @"public static class TestClass1 { }
-public class TestClass2 {
+
+public class TestClass2
+{
     public const int TestField1 = 1;
-    public static readonly int TestField2 = 2;
-    public static int TestField3;
-    public readonly int TestField4;
-    public int TestField5;
+    public static readonly int TestField2 = 1;
+    public static int TestField3 = 1;
+    public readonly int TestField4 = 1;
+    public int TestField5 = 1;
+    internal const int TestField6 = 1;
+    internal static readonly int TestField7 = 1;
+    internal static int TestField8 = 1;
+    internal readonly int TestField9 = 1;
+    internal int TestField10 = 1;
+    protected internal const int TestField11 = 1;
+    protected internal static readonly int TestField12 = 1;
+    protected internal static int TestField13 = 1;
+    protected internal readonly int TestField14 = 1;
+    protected internal int TestField15 = 1;
+    protected const int TestField16 = 1;
+    protected static readonly int TestField17 = 1;
+    protected static int TestField18 = 1;
+    protected readonly int TestField19 = 1;
+    protected int TestField20 = 1;
+    private const int TestField21 = 1;
+    private static readonly int TestField22 = 1;
+    private static int TestField23 = 1;
+    private readonly int TestField24 = 1;
+    private int TestField25 = 1;
+
+    public TestClass2()
```

Comment:
There is another case not covered here.

![image](https://cloud.githubusercontent.com/assets/10925800/9111608/a225c994-3c16-11e5-8037-d94ec05585ae.png)

A static class must be before a non-static one. Interestingly, this rule also applies to classes that are not nested. I suspect (though I didn't check) that the current implementation only works on children of a type.